src/order/antichain.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2020 Bhavik Mehta, Alena Gusakov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta, Alena Gusakov
+-/
+import order.basic
+import data.finset
+import data.multiset.finset_ops
+
+/-!
+# Antichains
+Investigating the structure of finsets in a partial order.
+We define antichains.
+
+## Main definitions
+* `antichain` is a finset of elements in a partial order where
+  no element is less than another.
+
+* The `join` of two antichains is the set obtained by taking
+  their union and removing all elements that are less than
+  another element.
+
+* The `meet` of two antichains is the set obtained by taking
+  their intersection and removing all elements that are less than
+  another element.
+-/
+
+open partial_order
+
+universe u
+
+open_locale classical
+noncomputable theory
+
+variable {α : Type u}
+
+section
+variables [partial_order α]
+
+/--
+A set of elements of a partial order forms an antichain if no two elements
+`A` and `B` are ordered `A < B`.
+-/
+def antichain (𝒜 : finset α) : Prop := ∀ a ∈ 𝒜, ∀ b ∈ 𝒜, a ≤ b → a = b
+
+lemma antichain_mem (A : finset α) (hA : antichain A) (a : α) :
+  a ∈ A → ∀ b ∈ A, a ≤ b → a = b :=
+begin
+  intros ha b hb hab,
+  exact hA a ha b hb hab
+end
+
+theorem subset_antichain (A B : finset α) (ha : antichain A) (hb : B ⊆ A) :
```

Comment:
This might work better as dot notation, so that you can use `ha.subset A B hb`
```suggestion
theorem antichain.subset (A B : finset α) (ha : antichain A) (hb : B ⊆ A) :
```